rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -30,7 +31,165 @@
+                            attribute=attribute,
+                            feature_type=feature_type,
+                            reg_lambda=reg_lambda,
+                        )
+                        self._tf_layers[name] = new_layer
+
+    @staticmethod
+    def _check_sparse_feature_decreased_sizes(
+        new_sparse_feature_sizes: List[int], old_sparse_feature_sizes: List[int],
+    ) -> None:
+        """Checks if the sizes of sparse features are decreased during fine-tuning.
+
+        Sparse feature sizes might decrease after changing the nlu file. This can
+        happen for example with `LexicalSyntacticFeaturizer`. We don't support
+        this behaviour and we raise a warning if this happens.
+
+        Args:
+            new_sparse_feature_sizes: sizes of current sparse features for a
+                                      specific attribute and feature type.
+            old_sparse_feature_sizes: sizes of sparse features the model was trained on
+                                      before for a specific attribute and feature type.
+        """
+        for new_size, old_size in zip(
+            new_sparse_feature_sizes, old_sparse_feature_sizes
+        ):
+            if new_size < old_size:
+                raise Exception(
+                    "Sparse feature sizes are decreased."
+                    "The NLU file was changed in a way that caused removing some"
+                    "features from certain featurizers (This might have been "
+                    "`LexicalSyntacticFeaturizer`). We don't support this behaviour"
+                    "at this point. One possible way to avoid this "
+                    "warning is retraining the model from scratch."
+                )
+
+    @staticmethod
+    def _create_dense_for_sparse_layer(
+        dense_layer: layers.DenseForSparse,
+        new_sparse_feature_sizes: List[int],
+        old_sparse_feature_sizes: List[int],
+        attribute: Text,
+        feature_type: Text,
+        reg_lambda: float,
+    ) -> layers.DenseForSparse:
+        """Creates a new `DenseForSparse` layer based on a given one.
+
+        Args:
+            dense_layer: a `DenseForSparse` layer that is used to create a new one.
+            new_sparse_feature_sizes: sizes of sparse features that will be
+                                      the input of the layer.
+            old_sparse_feature_sizes: sizes of sparse features that used to be
+                                      the input of the layer.
+            attribute: an attribute of the data fed to the layer.
+            feature_type: a feature type of the data fed to the layer.
+            reg_lambda: regularization constant.
+
+        Returns:
+            new `DenseForSparse` layer.
+        """
+        kernel = dense_layer.get_kernel().numpy()
+        use_bias, bias = dense_layer.get_bias_info()
+        if use_bias:
+            bias = bias.numpy()
```

Comment:
This is linked to my comment in the `DenseForSparse` layer class. Looks like you only need bias if `use_bias=True`. So that function can definitely be simplified to return only the bias if `use_bias` is `True`.